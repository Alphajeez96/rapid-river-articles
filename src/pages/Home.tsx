import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';

interface Article {
  body: string;
  id: number;
  title: string;
  userId: number;
  author: string;
  createdAt: number;
}

interface Author {
  name: string;
  id: number;
}

const CardSkeleton: React.FC = () => {
  return (
    <div className="feeds-card animate-pulse">
      <div className="bg-gray-300 h-72 w-full rounded-t-2xl"></div>
      <div className="p-4">
        {Array.from({ length: 3 }).map((_, index) => (
          <div key={index} className="h-3 bg-gray-300 my-4 rounded-sm"></div>
        ))}

        <div className="flex">
          <div className="h-3 w-24 bg-gray-300 my-4 rounded-sm"></div>
          <div className="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"></div>
        </div>
      </div>
    </div>
  );
};

const Card: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return (
    <div className="feeds-card">
      <img src="https://cdn.pixabay.com/photo/2017/09/23/16/11/compass-2779371_1280.jpg" alt="news" />

      {children}
    </div>
  );
};

const Home: React.FC = () => {
  let content;
  const [loading, setLoading] = useState(true);
  const [searchQuery, setsearchQuery] = useState('');
  const [articles, setArticles] = useState<Article[]>([]);
  const [filteredArticles, setFilteredArticles] = useState<Article[]>([]);

  const truncateText = (value: string, length: number = 50): string =>
    value && value.length > length ? value.substring(0, length) + '...' : value;

  const fetchArticles = async () => {
    try {
      const [articlesResponse, authorsResponse] = await Promise.all([
        fetch('https://jsonplaceholder.typicode.com/posts'),
        fetch('https://jsonplaceholder.typicode.com/users'),
      ]);

      const [articles, authors] = await Promise.all([articlesResponse.json(), authorsResponse.json()]);
      const mappedArticles: Article[] = articles.map((article: Article) => ({
        ...article,
        author: authors.find((author: Author) => author.id === article.userId)?.name || '',
        createdAt: Math.floor(Math.random() * 24) + 1,
      }));

      setArticles(mappedArticles);
    } catch (err) {
      console.error(err);
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    fetchArticles();
  }, []);

  useEffect(() => {
    const filtered = articles.filter(
      (article) =>
        article.title.toLowerCase().includes(searchQuery.toLowerCase()) ||
        article.author.toLowerCase().includes(searchQuery.toLowerCase()),
    );
    setFilteredArticles(filtered);
  }, [searchQuery, articles]);

  if (loading) {
    content = Array.from({ length: 6 }).map((_, index) => <CardSkeleton key={index} />);
  } else if (filteredArticles.length) {
    content = filteredArticles.map((article: Article) => (
      <Card key={article.id}>
        <div className="p-4">
          <p className="text"> {truncateText(article.title)}</p>
          <p className="text pt-4">{truncateText(article.body, 150)}</p>

          <div className="flex items-center pt-4">
            <p className="text-sm font-semibold text-primary-grey">{article.author}</p>

            <div className="ml-auto flex items-center gap-2">
              <span className="time-dot"></span>
              <p className="text-xs font-medium text-primary-grey">
                {' '}
                {`${article.createdAt} ${article.createdAt > 1 ? 'hours' : 'hour'} ago`}{' '}
              </p>
            </div>
          </div>
        </div>
      </Card>
    ));
  } else {
    content = <p className="empty-text">No articles found</p>;
  }

  return (
    <section className="feeds-holder">
      <div className="flex">
        <aside className="flex items-center gap-3 w-1/3">
          <div className="relative w-full">
            <input
              type="text"
              placeholder="Search by author or title"
              className="input search-input"
              value={searchQuery}
              onChange={(e) => setsearchQuery(e.target.value)}
            />
            <span className="absolute top-[10px] left-4">
              <img src="/src/assets/images/search.svg" alt="search-icon" />
            </span>
          </div>
        </aside>

        <div className="ml-auto">
          {' '}
          <Link to="/new">
            {' '}
            <button className="button">Create article</button>{' '}
          </Link>
        </div>
      </div>

      <div className="inner-holder transparent-scrollbar">{content}</div>
    </section>
  );
};

export default Home;
