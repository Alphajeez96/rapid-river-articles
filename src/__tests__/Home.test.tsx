/* eslint-disable @typescript-eslint/no-explicit-any */
import { describe, expect, test, vi } from 'vitest';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../pages/Home';

window.fetch = vi.fn();

function createFetchResponse(data: any) {
  return { json: () => new Promise((resolve) => resolve(data)) };
}

describe('Renders the Home page', () => {
  test('snapshot', () => {
    const comp = render(
      <Router>
        <Home />
      </Router>,
    );
    expect(comp).toMatchSnapshot();
  });

  test('fetches articles', async () => {
    vi.useFakeTimers();
    const mockArticles = [
      {
        id: 1,
        title: 'Test Article',
        body: 'This is a test article.',
        userId: 1,
      },
    ];

    window.fetch.mockResolvedValueOnce(createFetchResponse(mockArticles));
    render(
      <Router>
        <Home />
      </Router>,
    );

    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith('https://jsonplaceholder.typicode.com/posts');
  });
});
