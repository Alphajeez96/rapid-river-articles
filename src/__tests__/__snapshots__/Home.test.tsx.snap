// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders the Home page > snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="feeds-holder"
      >
        <div
          class="flex"
        >
          <aside
            class="flex items-center gap-3 w-1/3"
          >
            <div
              class="relative w-full"
            >
              <input
                class="input search-input"
                placeholder="Search by author or title"
                type="text"
                value=""
              />
              <span
                class="absolute top-[10px] left-4"
              >
                <img
                  alt="search-icon"
                  src="/src/assets/images/search.svg"
                />
              </span>
            </div>
          </aside>
          <div
            class="ml-auto"
          >
             
            <a
              href="/new"
            >
               
              <button
                class="button"
              >
                Create article
              </button>
               
            </a>
          </div>
        </div>
        <div
          class="inner-holder transparent-scrollbar"
        >
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="feeds-holder"
    >
      <div
        class="flex"
      >
        <aside
          class="flex items-center gap-3 w-1/3"
        >
          <div
            class="relative w-full"
          >
            <input
              class="input search-input"
              placeholder="Search by author or title"
              type="text"
              value=""
            />
            <span
              class="absolute top-[10px] left-4"
            >
              <img
                alt="search-icon"
                src="/src/assets/images/search.svg"
              />
            </span>
          </div>
        </aside>
        <div
          class="ml-auto"
        >
           
          <a
            href="/new"
          >
             
            <button
              class="button"
            >
              Create article
            </button>
             
          </a>
        </div>
      </div>
      <div
        class="inner-holder transparent-scrollbar"
      >
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`The component UI should not change 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="feeds-holder"
      >
        <div
          class="flex"
        >
          <aside
            class="flex items-center gap-3 w-1/3"
          >
            <div
              class="relative w-full"
            >
              <input
                class="input search-input"
                placeholder="Search by author or title"
                type="text"
                value=""
              />
              <span
                class="absolute top-[10px] left-4"
              >
                <img
                  alt="search-icon"
                  src="/src/assets/images/search.svg"
                />
              </span>
            </div>
          </aside>
          <div
            class="ml-auto"
          >
             
          </div>
        </div>
        <div
          class="inner-holder transparent-scrollbar"
        >
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
          <div
            class="feeds-card animate-pulse"
          >
            <div
              class="bg-gray-300 h-72 w-full rounded-t-2xl"
            />
            <div
              class="p-4"
            >
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="h-3 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="flex"
              >
                <div
                  class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
                />
                <div
                  class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="feeds-holder"
    >
      <div
        class="flex"
      >
        <aside
          class="flex items-center gap-3 w-1/3"
        >
          <div
            class="relative w-full"
          >
            <input
              class="input search-input"
              placeholder="Search by author or title"
              type="text"
              value=""
            />
            <span
              class="absolute top-[10px] left-4"
            >
              <img
                alt="search-icon"
                src="/src/assets/images/search.svg"
              />
            </span>
          </div>
        </aside>
        <div
          class="ml-auto"
        >
           
        </div>
      </div>
      <div
        class="inner-holder transparent-scrollbar"
      >
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
        <div
          class="feeds-card animate-pulse"
        >
          <div
            class="bg-gray-300 h-72 w-full rounded-t-2xl"
          />
          <div
            class="p-4"
          >
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="h-3 bg-gray-300 my-4 rounded-sm"
            />
            <div
              class="flex"
            >
              <div
                class="h-3 w-24 bg-gray-300 my-4 rounded-sm"
              />
              <div
                class="ml-auto h-3 w-20 bg-gray-300 my-4 rounded-sm"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
